platforms/core-execution/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonServer.java

```diff
@@ -213,7 +213,7 @@ public static CloseableServiceRegistry create(ServiceRegistry parent, File baseD
                 .displayName("worker request services for " + baseDir.getAbsolutePath())
                 .parent(parent)
                 .provider(new WorkerProjectServices())
-                .provider(new WorkerSharedProjectScopeServices(baseDir))
+                .provider(new WorkerSharedProjectScopeServices(baseDir, baseDir)) // TODO: needs proper settings dir
```

Comment:
❌  Needs to be addressed

Reply:
I'm quite unsure how... Ongoing, but ideas welcome.

Reply:
`ProjectLayout` can be injected in worker actions.
We need to pass the settings directory down to here, across the wire for process isolated workers.
This will have the effect of also passing it to our internal workers for compiler daemons, which is fine.